nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,799 @@
+        The strategy used to compute the mask: use 'background' if your
+        images present a clear homogeneous background, and 'epi' if they
+        are raw EPI images. Depending on this value, the mask will be
+        computed from masking.compute_background_mask or
+        masking.compute_epi_mask. Default is 'background'.
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional. Default: 0
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs : int, optional. Default: 1.
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs'.
+
+    verbose : int, optional. Default: False.
+        Verbosity level.
+    """
+    def __init__(self, estimator='svc', mask=None, cv=10, param_grid=None,
+                 screening_percentile=20, scoring=None, smoothing_fwhm=None,
+                 standardize=True, target_affine=None, target_shape=None,
+                 low_pass=None, high_pass=None, t_r=None,
+                 mask_strategy='background', is_classif=True, memory=None,
+                 memory_level=0, n_jobs=1, verbose=False):
+        self.estimator = estimator
+        self.mask = mask
+        self.cv = cv
+        self.param_grid = param_grid
+        self.screening_percentile = screening_percentile
+        self.scoring = scoring
+        self.is_classif = is_classif
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def fit(self, X, y):
+        """Fit the decoder (learner).
+
+        Parameters
+        ----------
+        X : list of Niimg-like objects
+            See http://nilearn.github.io/manipulating_images/input_output.html
+            Data on which model is to be fitted. If this is a list,
+            the affine is considered the same for all.
+
+        y : array or list of length n_samples
+            The dependent variable (age, sex, QI, etc.).
+            Target variable to predict. Must have exactly as many elements as
+            3D images in niimg.
+
+        Attributes
+        ----------
+        `masker_` : instance of NiftiMasker
+            The nifti masker used to mask the data.
+
+        `mask_img_`: NiImage
+            Mask computed by the masker object.
+
+        `classes_`: numpy.ndarray
+            Classes to predict. For classification only.
+
+        `screening_percentile_` : float
```

Comment:
Some have with space : and some without space : 
May be we keep same at all places for consistency.